docs/typescript/utility-and-manipulation.md

```diff
@@ -0,0 +1,84 @@
+# Utility types and manipulation
+
+## Utility types
+Common type transformations that make them easier to write. You can find more on [TS docs](https://www.typescriptlang.org/docs/handbook/utility-types.html).
+
+### `Partial<Type>` – Makes all properties `optional`
+```ts
+interface Todo {
+  title: string;
+  description: string;
+  completed: boolean;
+}
+
+type TodoUpdate = Partial<Todo>
```

Comment:
Is there autoformatting applied to markdown? I think WordPress flavored `prettier` would include spaces inside the angle brackets. We're not consistent across files here, let's make sure we are.

```suggestion
type TodoUpdate = Partial< Todo >;
```

Reply:
I'm finding that VSCode prettier will use the repo's `.prettierrc` for auto-formatting `.md`.

`singleQuote: true` and `semi: true` rules should be applied to this file in addition to `parenSpacing`.

Reply:
I ran prettier on the markdown files in 30ec8452a56a410d0e39f486ecf99c956ea900a4 so the formatting _should_ be consistent now.